### PR TITLE
Add support of test-only criterions after each epoch

### PR DESCRIPTION
**Pull request by Emrys365**
This PR enables specifying whether some criterions in the ENH task will only be calculated during inference (after each epoch).
This can be now achieved by adding an additional argument `only_for_test` in the model config file:
```yaml
criterions: 
  # The first criterion
  - name: si_snr 
    conf:
      eps: 1.0e-7
      only_for_test: True
    # the wrapper for the current criterion
    # for single-talker case, we simplely use fixed_order wrapper
    wrapper: fixed_order
    wrapper_conf:
      weight: 1.0
```

### CODE REVIEW

File: espnet2/enh/loss/criterions/abs_loss.py [hunk 1]
```diff
@@ -13,6 +13,12 @@ class AbsEnhLoss(torch.nn.Module, ABC):
     def name(self) -> str:
         return NotImplementedError
 
+    # This property specifies whether the criterion will only
+    # be evaluated during the inference stage
+    @property
+    def only_for_test(self) -> bool:
+        return NotImplementedError
```
Comment (LiChenda) [inline espnet2/enh/loss/criterions/abs_loss.py:20]: Given that most losses are used in both training and testing, maybe we can set defalut `False` here?

**Review by LiChenda — COMMENTED** [review]
LGTM me!

**Comment by sw005320** [comment]
Thanks a lot, @Emrys365!
